packages/mds-ingest-service/service/provider.ts

```diff
@@ -215,19 +245,8 @@ export const IngestServiceProvider: ServiceProvider<IngestService & IngestMigrat
 
   writeMigratedTelemetry: async (telemetry, migrated_from) => {
     try {
-      const [model = null] = await IngestRepository.writeMigratedTelemetry([telemetry], migrated_from)
-      if (model) {
-        const [cached, streamed] = await Promise.allSettled([
-          cache.writeTelemetry([model]),
-          stream.writeTelemetry([model])
-        ])
-        if (cached.status === 'rejected') {
-          IngestServiceLogger.warn('Error writing telemetry to cache', { telemetry: model, error: cached.reason })
-        }
-        if (streamed.status === 'rejected') {
-          IngestServiceLogger.warn('Error writing telemetry to stream', { telemetry: model, error: streamed.reason })
-        }
-      }
+      const model = await writeMigratedTelemetry(telemetry, migrated_from)
```

Comment:
So, these things _can_ return `null` because the repository methods have `ON CONFLICT DO NOTHING` clauses which will not return an entity if one isn't inserted.